staging/src/k8s.io/apiserver/pkg/cel/mutation/optional_test.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mutation
+
+import (
+	"reflect"
+	"strings"
+	"testing"
+)
+
+// TestCELOptional is an exploration test to demonstrate how CEL optional library
+// behave for the use cases that the mutation library requires.
+func TestCELOptional(t *testing.T) {
+	for _, tc := range []struct {
+		name                 string
+		expression           string
+		expectedValue        any
+		expectedCompileError string
+	}{
+		{
+			// question mark syntax still requires the field to exist in object construction
```

Comment:
The behavior of optional seems different between object and map. For object, optional will fail compilation if field not exist. This is worth noting.. Maybe also add more tests here to show the expected behavior?

Reply:
Added comments to the tests to explain these. Thank you.